pkg/network/certificates_knobs.go

```diff
@@ -0,0 +1,99 @@
+package network
+
+import (
+	"reflect"
+	"time"
+
+	"github.com/pkg/errors"
+
+	opv1alpha1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+)
+
+// validateCertificateKnobs validates the following fields
+// - CARotateInterval
+// - CAOverlapInterval
+// - CertRotateInterval
+func validateCertificateKnobs(conf *opv1alpha1.NetworkAddonsConfigSpec) []error {
+	if conf.CertificateKnobs == nil {
+		return []error{}
+	}
+
+	certificateKnobs := *conf.CertificateKnobs
+
+	errs := []error{}
+
+	caRotateInterval := parseCertificateKnob("caRotateInterval", certificateKnobs.CARotateInterval, &errs)
+	caOverlapInterval := parseCertificateKnob("caOverlapInterval", certificateKnobs.CAOverlapInterval, &errs)
+	certRotateInterval := parseCertificateKnob("certRotateInterval", certificateKnobs.CertRotateInterval, &errs)
+
+	// If they cannot be parsed don't continue
+	if len(errs) > 0 {
+		return errs
+	}
+
+	validateGreaterThanZero("caRotateInterval", caRotateInterval, &errs)
+	validateGreaterThanZero("caOverlapInterval", caOverlapInterval, &errs)
+	validateGreaterThanZero("certRotateInterval", certRotateInterval, &errs)
+
+	// If we have a zero value don't continue
+	if len(errs) > 0 {
+		return errs
+	}
+
+	validateGreaterThan("caRotateInterval", caRotateInterval, "caOverlapInterval", caOverlapInterval, &errs)
+	validateGreaterThan("caRotateInterval", caRotateInterval, "certRotateInterval", certRotateInterval, &errs)
+
+	return errs
+}
+
+func fillDefaultsCertificateKnobs(conf, previous *opv1alpha1.NetworkAddonsConfigSpec) []error {
+	if conf.CertificateKnobs == nil {
+		return []error{}
```

Comment:
This functions should fill in defaults. Could we keep them here? Something close to the current KubeVirt defaults. Otherwise we will have undefined behavior where everything is controlled by the component, I don't think that is desirable.

Reply:
the kube-admission-webhook have some defaults, but we can put here the ones like kubevirt.

Reply:
Done.